pysnmp/proto/secmod/rfc3414/service.py

```diff
@@ -333,35 +333,40 @@ def __generateRequestOrResponseMsg(self, snmpEngine,
             0, scopedPDU, verifyConstraints=False, matchTags=False, matchConstraints=False
         )
 
-        # 3.1.6a
-        if securityStateReference is None and securityLevel in (2, 3):
-            if securityEngineID in self.__timeline:
-                (snmpEngineBoots, snmpEngineTime, latestReceivedEngineTime,
+        snmpEngineBoots = snmpEngineTime = 0
+
+        if securityLevel in (2, 3):
```

Comment:
The securityLevel determine statements is moved to the outside. This may cause an issue: when receiving a get request from non-authoritative SNMP engine which is not authenticated and encrypted, the boots/time in the reply message will always be zero

Reply:
I think this should be OK because boots/time values are only used in the crypto context.

Also, before this patch the behaviors [seems to be the same](https://github.com/etingof/pysnmp/pull/152/files/6240bcbfbcb20c5927030950817b267039d334aa#diff-e2fca5ba33a8d3926bf1535759b5e50eL337), no?

Reply:
Before this patch, boots/time are read from LCD when sending the response message regardless of the securityLevel. I think this is based on rfc2574 3.1.6.b, and it's useful when a non-authoritative SNMP engine send a discovery request message to learn the authoritative SNMP engine's information.

Also I insert some debug log of the response to noAuthNoPriv discovery request message.

 04:05:06,460 pysnmp: processIncomingMsg: cache write securityStateReference 2089007 by msgUserName 
 04:05:06,460 pysnmp: processIncomingMsg: unknown securityEngineID OctetString('')
 04:05:06,460 pysnmp: processIncomingMsg: null or malformed msgAuthoritativeEngineID
 04:05:06,460 pysnmp: processIncomingMsg: starting snmpEngineID discovery procedure
 04:05:06,462 pysnmp: StatusInformation: {'securityLevel': 1, 'contextName': OctetString('CloudMCU'), 'val': Counter32(7), 'contextEngineId': OctetString(''), 'oid': (1, 3, 6, 1, 6, 3, 15, 1, 1, 4, 0), 'scopedPDU': ScopedPDU(componentType=NamedTypes(NamedType('contextEngineId', OctetString()), NamedType('contextName', OctetString()), NamedType('data', PDUs(componentType=NamedTypes(NamedType('get-request', GetRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('get-next-request', GetNextRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('get-bulk-request', GetBulkRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('non-repeaters', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('max-repetitions', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('response', ResponsePDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('set-request', SetRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('inform-request', InformRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('snmpV2-trap', SNMPv2TrapPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('report', ReportPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList()))))))))).setComponents(OctetString(''), OctetString('CloudMCU'), PDUs(componentType=NamedTypes(NamedType('get-request', GetRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('get-next-request', GetNextRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('get-bulk-request', GetBulkRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('non-repeaters', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('max-repetitions', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('response', ResponsePDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('set-request', SetRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('inform-request', InformRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('snmpV2-trap', SNMPv2TrapPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('report', ReportPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))))).setComponents(GetRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList()))).setComponents(Integer32(1475743822), Integer('noError', NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18))), Integer(0, subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647)))), VarBindList()))), 'maxSizeResponseScopedPDU': 65443, 'securityStateReference': 2089007, 'errorIndication': <pysnmp.proto.errind.UnknownEngineID instance at 0x7f3f5f459368>}
 04:05:06,464 pysnmp: prepareDataElements: SM failed, statusInformation {'securityLevel': 1, 'contextName': OctetString('CloudMCU'), 'val': Counter32(7), 'contextEngineId': OctetString(''), 'oid': (1, 3, 6, 1, 6, 3, 15, 1, 1, 4, 0), 'scopedPDU': ScopedPDU(componentType=NamedTypes(NamedType('contextEngineId', OctetString()), NamedType('contextName', OctetString()), NamedType('data', PDUs(componentType=NamedTypes(NamedType('get-request', GetRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('get-next-request', GetNextRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('get-bulk-request', GetBulkRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('non-repeaters', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('max-repetitions', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('response', ResponsePDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('set-request', SetRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('inform-request', InformRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('snmpV2-trap', SNMPv2TrapPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('report', ReportPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList()))))))))).setComponents(OctetString(''), OctetString('CloudMCU'), PDUs(componentType=NamedTypes(NamedType('get-request', GetRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('get-next-request', GetNextRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('get-bulk-request', GetBulkRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('non-repeaters', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('max-repetitions', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('response', ResponsePDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('set-request', SetRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('inform-request', InformRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('snmpV2-trap', SNMPv2TrapPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('report', ReportPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))))).setComponents(GetRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList()))).setComponents(Integer32(1475743822), Integer('noError', NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18))), Integer(0, subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647)))), VarBindList()))), 'maxSizeResponseScopedPDU': 65443, 'securityStateReference': 2089007, 'errorIndication': <pysnmp.proto.errind.UnknownEngineID instance at 0x7f3f5f459368>}
 04:05:06,465 pysnmp: prepareResponseMessage: stateReference 1803105
 04:05:06,466 pysnmp: prepareResponseMessage: prepare report PDU for statusInformation {'securityLevel': 1, 'contextName': OctetString('CloudMCU'), 'val': Counter32(7), 'contextEngineId': OctetString(''), 'oid': (1, 3, 6, 1, 6, 3, 15, 1, 1, 4, 0), 'scopedPDU': ScopedPDU(componentType=NamedTypes(NamedType('contextEngineId', OctetString()), NamedType('contextName', OctetString()), NamedType('data', PDUs(componentType=NamedTypes(NamedType('get-request', GetRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('get-next-request', GetNextRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('get-bulk-request', GetBulkRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('non-repeaters', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('max-repetitions', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('response', ResponsePDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('set-request', SetRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('inform-request', InformRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('snmpV2-trap', SNMPv2TrapPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('report', ReportPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList()))))))))).setComponents(OctetString(''), OctetString('CloudMCU'), PDUs(componentType=NamedTypes(NamedType('get-request', GetRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('get-next-request', GetNextRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('get-bulk-request', GetBulkRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('non-repeaters', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('max-repetitions', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('response', ResponsePDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('set-request', SetRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('inform-request', InformRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('snmpV2-trap', SNMPv2TrapPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))), NamedType('report', ReportPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList())))))).setComponents(GetRequestPDU(componentType=NamedTypes(NamedType('request-id', Integer32()), NamedType('error-status', Integer(, NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18)))), NamedType('error-index', Integer(subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647))))), NamedType('variable-bindings', VarBindList()))).setComponents(Integer32(1475743822), Integer('noError', NamedValues(('noError', 0), ('tooBig', 1), ('noSuchName', 2), ('badValue', 3), ('readOnly', 4), ('genErr', 5), ('noAccess', 6), ('wrongType', 7), ('wrongLength', 8), ('wrongEncoding', 9), ('wrongValue', 10), ('noCreation', 11), ('inconsistentValue', 12), ('resourceUnavailable', 13), ('commitFailed', 14), ('undoFailed', 15), ('authorizationError', 16), ('notWritable', 17), ('inconsistentName', 18))), Integer(0, subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(0, Integer(2147483647)))), VarBindList()))), 'maxSizeResponseScopedPDU': 65443, 'securityStateReference': 2089007, 'errorIndication': <pysnmp.proto.errind.UnknownEngineID instance at 0x7f3f5f459368>}
 04:05:06,467 pysnmp: prepareResponseMessage: using contextEngineId SnmpEngineID(hexValue='8000ea2ffc4a1e86'), contextName OctetString('CloudMCU')
 04:05:06,468 pysnmp: prepareResponseMessage: SNMPv3Message:
 msgVersion=3
 msgGlobalData=HeaderData:
  msgID=1475651699
  msgMaxSize=65507
  msgFlags=0x00
  msgSecurityModel=3

 msgSecurityParameters=0x302e04088000ea2ffc4a1e86020102020121040475736572040c0000000000000000000000000408d4c0c68b84c8243a
 msgData=ScopedPduData:
  encryptedPDU=0x4f3c7e583b81486e0994ce16ab4f3f5c45b4327b0e9b280c7de1c2931cffdc2baafcc021b6e3276620abae2adea15f35051da3ae112a3da6fb5d3aff6b9d001f75d19f104cac5ef80f9a1c10c959f26b568bccaa1e74c84b908c448e82bebc9d


 04:05:06,468 pysnmp: prepareResponseMessage: securityModel Integer(3, subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueRangeConstraint(1, 2147483647))), securityEngineId SnmpEngineID(hexValue='8000ea2ffc4a1e86'), securityName None, securityLevel 1
 04:05:06,468 pysnmp: __generateRequestOrResponseMsg: user info read from cache
 04:05:06,468 pysnmp: __generateRequestOrResponseMsg: local usmUserName OctetString('', subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueSizeConstraint(0, 32))) usmUserSecurityName OctetString('', subtypeSpec=ConstraintsIntersection(ConstraintsIntersection(), ValueSizeConstraint(0, 32))) usmUserAuthProtocol (1, 3, 6, 1, 6, 3, 10, 1, 1, 1) usmUserPrivProtocol (1, 3, 6, 1, 6, 3, 10, 1, 2, 1) securityEngineID SnmpEngineID(hexValue='8000ea2ffc4a1e86') securityName None
 04:05:06,468 pysnmp: __generateRequestOrResponseMsg: read snmpEngineBoots, snmpEngineTime from LCD
 04:05:06,468 pysnmp: __generateRequestOrResponseMsg: use snmpEngineBoots 2 snmpEngineTime 72 for securityEngineID SnmpEngineID(hexValue='8000ea2ffc4a1e86')